### DISCUSSION TITLE
Generic reusable type for parameter t

### DISCUSSION THREAD

**Discussion by dylanlarrivee** [hayes/pothos#927] [Q&A] [answered]
Hi All,
I was hoping to use the `noImplicitAny: true` setting in my tsconfig but doing so requires a type to be provided for the t parameter that is used throughout my Pothos code. 

I thought I saw in a discussion earlier that someone had posted a reply with a reusable generic type they had created and was hoping it could be shared again with me as I have been unable to find it again.

Thanks,

**Comment by hayes** [comment]
Do you have details of the error you are seeing? I always use use no implicit any, and have never had an issue (t should definitely not be typed as any unless something is misconfigured)

**Reply by dylanlarrivee** [reply]
When compiling my code I get this error: `#14 48.73 src/modules/alert/entities/Alert.ts(4,12): error TS7006: Parameter 't' implicitly has an 'any' type.`

Code it is referencing:
`builder.prismaObject('Alert', {
  fields: (t) => ({`

**Reply by hayes** [reply, marked as answer]
This is likely an issue with how you have set up the Prisma plugin, if you open the file you import the PrismaTypes from, have they been generated (does the file exists, and does it have a bunch of types in it)

**Reply by dylanlarrivee** [reply]
Ok cool, thanks for confirming. This is happening during a Docker build and it looks like there is an issue with my prisma generate step which would explain the missing types.

**Reply by dylanlarrivee** [reply]
Yep, looks like it was a mix of that and a misconfiguration of my tsconfig file as I am trying to build out docker containers for a monorepo setup.
